### PR TITLE
[OOE-50] Add the ability to generate a report link

### PR DESCRIPTION
**Pull request by DFieldFL**

### CODE REVIEW

**Comment by DFieldFL** [comment]
@takipi/keymaker please review

File: src/main/java/com/overops/plugins/bamboo/TaskType.java [hunk 1]
```diff
@@ -70,27 +67,43 @@ public TaskResult execute(@NotNull TaskContext context) throws TaskException {
...
+            if (displayLink)
+            {
+                String appUrl = (String)globalSettings.get(Const.GLOBAL_APP_URL);
+                HtmlParts htmlParts = new HtmlParts(overOpsService.generateReportLinkHtml(appUrl, query), "");
+                context.getBuildContext().getBuildResult().getCustomBuildData().put("overOpsReport", objectMapper.writeValueAsString(htmlParts));
+                context.getBuildContext().getBuildResult().getCustomBuildData().put("isOverOpsStep", "true");
+                return resultBuilder.success().build();
+            } else
+            {
+                logger.addBuildLogEntry("[" + Utils.getArtifactId() + " v" + Utils.getVersion() + "]");
```
Comment (daverted) [inline src/main/java/com/overops/plugins/bamboo/TaskType.java]: do we want to log the plugin version regardless of the state of `displayLink`?

File: src/main/java/com/overops/plugins/bamboo/TaskType.java [hunk 2]
```diff
@@ -70,27 +67,43 @@ public TaskResult execute(@NotNull TaskContext context) throws TaskException {
...
+
+                boolean showAllEvents = Boolean.parseBoolean(context.getConfigurationMap().get(Const.SHOW_ALL_EVENTS));
+                boolean isDebug = Boolean.parseBoolean(context.getConfigurationMap().get(Const.DEBUG));
+                PrintStream printStream = isDebug ? new BambooPrintWriter(System.out, logger) : null;
```
Comment (daverted) [inline src/main/java/com/overops/plugins/bamboo/TaskType.java]: I think this should move outside `if (displayLink)` as well.. we probably want to have debug logging regardless of the state of this variable to check for other settings such as API host, env ID, etc.

**Review by daverted — CHANGES_REQUESTED** [review]
See comments around debug logging. I see that the quality report has an `isDebug` flag - is it possible to show any debugging when we're generating the link and not the full report? It has been useful in the past to have some debugging available to verify basic settings like API URL, env ID, critical exception types, etc. just to make sure there are no typos and settings which can be set in multiple places are being picked up as expected.

File: src/main/java/com/overops/plugins/bamboo/TaskType.java (hunk 1 above)
Comment (DFieldFL) [reply src/main/java/com/overops/plugins/bamboo/TaskType.java]: 👍

File: src/main/java/com/overops/plugins/bamboo/TaskType.java (hunk 2 above)
Comment (DFieldFL) [reply src/main/java/com/overops/plugins/bamboo/TaskType.java]: Updated to show the major settings.

**Review by daverted — APPROVED** [review]

**Pull request merged**